tsl/test/sql/continuous_aggs_union_view.sql

```diff
@@ -0,0 +1,119 @@
+SELECT pg_get_viewdef('metrics_summary',true);
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- refresh
+REFRESH MATERIALIZED VIEW metrics_summary;
+-- result should not change after refresh for union view
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- downgrade view to non-union view
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=true);
+-- this should be view without union
+SELECT pg_get_viewdef('metrics_summary',true);
+-- view should have results now after refresh
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+DROP VIEW metrics_summary CASCADE;
+
+-- check default view for new continuous aggregate with materialized_only to true
+CREATE VIEW metrics_summary WITH (timescaledb.continuous, timescaledb.materialized_only=true) AS SELECT time_bucket('1d',time), avg(value) FROM metrics GROUP BY 1;
+
+-- this should be view without union
+SELECT pg_get_viewdef('metrics_summary',true);
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- upgrade view to union view
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=false);
+-- this should be union view
+SELECT pg_get_viewdef('metrics_summary',true);
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- downgrade view to non-union view
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=true);
+-- this should be view without union
+SELECT pg_get_viewdef('metrics_summary',true);
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+DROP VIEW metrics_summary CASCADE;
+
+--
+-- test queries on union view
+--
+
+CREATE VIEW metrics_summary WITH (timescaledb.continuous, timescaledb.materialized_only=true) AS SELECT time_bucket('1d',time), avg(value) FROM metrics GROUP BY 1;
+
+-- query should not have results since cagg is materialized only and no refresh has happened yet
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=false);
+
+-- after switch to union view all results should be returned
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+REFRESH MATERIALIZED VIEW metrics_summary;
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=true);
+
+-- materialized only view should return data now too because refresh has happened
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- add some more data
+INSERT INTO metrics(time, device_id, value) SELECT '2000-02-01'::timestamptz, 1, 0.1;
+INSERT INTO metrics(time, device_id, value) SELECT '2000-02-01'::timestamptz, 2, 0.2;
+INSERT INTO metrics(time, device_id, value) SELECT '2000-02-01'::timestamptz, 3, 0.3;
+
+-- materialized only view should not have new data yet
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- but union view should
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=false);
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+
+-- and after refresh non union view should have new data too
```

Comment:
1) can you add a test with multiple cont agg views on the same table : 1 is materialized = true and the other false (might be good for continuous_aggs_usage to demonstarte the difference in behavior).
2) cagg view with  default materialized_only (false)  + ignore_invalidations setting.
and new data is added after the REFRESH.
3) what happens with materialized_only = false and the cagg dropped old chunks but the raw data still exists. Could you add a test for that?
Instead of default chunk interval, could you set the chunks specifically so we know that we have a) multiple chunks in the cont . agg table and b) when raw chunks are dropped, we still have some partially filled chunks in mat table.

Reply:
if you drop chunks from the cagg older than watermark they will disappear from the output, until you run REFRESH to rebuild them, since the mode to honor invalidations did not get added